src/threads/table.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { dynamodb, toDBItem } from '../dynamodb';
+import { ForumTable } from './table';
+
+const putItem = (threadId: string, timestamp: number, data: Record<string, unknown>) => dynamodb.putItem({
+  TableName: 'forumTable',
+  Item: toDBItem({ threadId, timestamp, ...data }),
+});
+const getAll = (threadId: string) => dynamodb.query({
+  TableName: 'forumTable',
+  ExpressionAttributeValues: { ':tid': { S: threadId } },
+  KeyConditionExpression: 'threadId = :tid',
+});
+const deleteAllItems = async (threadId: string) => {
+  const result = await getAll(threadId);
+  await Promise.all((result.Items || []).map(item => dynamodb.deleteItem({ TableName: 'forumTable', Key: item })));
+};
+
+describe('Forum table', () => {
+  const forumTable = new ForumTable(dynamodb);
+  const consoleErrorStub = jest.spyOn(global.console, 'error');
+  consoleErrorStub.mockImplementation(() => {});
+
+  describe('getThreadEvents()', () => {
+    const participantId = 'participantId';
+    const itemId = 'itemId';
+    const threadId = `THREAD#${participantId}#${itemId}`;
+
+    afterEach(async () => {
+      await deleteAllItems(threadId);
+    });
+
+    it('should fail at decode step', async () => {
+      await putItem(threadId, 1, { type: 'unknown_type' });
+      await expect(forumTable.getThreadEvents(participantId, itemId)).rejects.toBeInstanceOf(Error);
+    });
```

Comment:
There are also other error case that worth testing. The db in error, a "thread_opened" with a non-number timestamp, a "thread_opened" with a no or invalid "byUserId", ...

Reply:
Db in error will result in an error given by the aws lib, testing it means testing aws lib.
"thread_opened" with a non number timestamp is impossible since we provide the timestamp ourselves: `Date.now()`
"thread_opened" with no or invalid user id is impossible due to typescript guards, it's redundant to test something backed by typescript.